app/Http/Controllers/Api/MaterialController.php

```diff
@@ -57,12 +57,23 @@ public function update(Request $request, $id)
     public function destroy($id)
     {
         $material = Material::find($id);
+
         if (!$material) {
             return response()->json([
                 'error' => "Material doesn't exists"
             ], 400);
+        } else {
+            // The != was used because in tests the amount is not being perceived as integer
+            if ($material->amount != 0) {
+                --$material->amount;
+                $material->save();
+
+                return response()->json($material, 200);
+            }
```

Comment:
It doesn't make sense, because this operation should be done in LoanController, when it to loan a material. The destroy function should only delete a material.

Reply:
But if a material is broken and I have five more of the same type? I will have to delete the five materials to remove the broken one?

Reply:
> I noticed that you have included the commits of previous PR. In other words, you violated the git flow life circle.

Sorry man, the change would be so tiny that I didn't wanted to create a new pull request for it. But I will keep that in mind in the next time.

Reply:
> But if a material is broken and I have five more of the same type? I will have to delete the five materials to remove the broken one?

So you will update a material to remove one of the same. The destroy function it's to last case when an user wants delete a material completely.

Reply:
Oh, now I get it, thanks.